lib/board.rb

```diff
@@ -0,0 +1,153 @@
+
+class Board
+  attr_accessor :board
+  def initialize(size=10)
+    @size = size
+    @board = Array.new(@size) {Array.new(@size){Cell.new}}
+    place_mines(generate_mines)
+    set_all_cell_adjacent_mines
+
+  end
+
+  def generate_mines
+    (0..((@size*@size)-1)).to_a.sample(@size)
+  end
+
+  def place_mines mine_locations
+    mine_locations.each do |x|
+      tens = x / 10
+      ones = x % 10
+      @board[tens][ones].mine = true
+    end
+  end
```

Comment:
Interesting way to do it! Eliminates the need to check if a randomly selected location already has a mine. You might prefer `@size**2 - 1` on line 13.